Add recceiver id, which is used by cleanOnStart and cleanOnStop

jacomago opened the pull request.

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -521,6 +512,15 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocid, owner, io
+                        {u'name': 'pvStatus', u'owner': owner, u'value': 'Active'},
+                        {u'name': 'time', u'owner': owner, u'value': iocTime},
+                        {u'name': RECCEIVER_ID_KEY, u'owner': owner, u'value': recceiver_id}]
+
```
While I do really like these cleanups, it might have been better to put them in a separate commit to make it clear what is cleanup, and what is the functional change.

jacomago replied:
done

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -467,7 +457,8 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocid, owner, io
                      {u'name': 'pvStatus', u'owner': owner, u'value': "Active"},
-                     {u'name': 'time', u'owner': owner, u'value': iocTime}]
+                     {u'name': 'time', u'owner': owner, u'value': iocTime},
+                     {u'name': RECCEIVER_ID_KEY, u'owner': owner, u'value': recceiver_id}]
```
Given your new function `create_properties` has a lot of overlap with this, isn't it possible to use it here as well?

jacomago replied:
done

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -224,7 +227,7 @@ def _commitWithThread(self, TR):
         for rid, (info) in pvInfo.items():
             if info["pvName"] in pvInfoByName:
-                _log.warn("Commit contains multiple records with PV name: %s (%s)", pv, iocid)
+                _log.warn("Commit contains multiple records with PV name: %s (%s)", info["pvName"], iocid)
```
Good catch...

simon-ess reviewed (commented):
Broadly speaking this seems like a good change. My one main concern is that leaving it up to the user to configure the RECCEIVER_ID seems potentially problematic. What happens if multiple recceivers share a key? Is there no way to use something like `hostname` or something similar to specify the ID?

jacomago committed:
Clean up duplication

Creation of small methods:
  - remove_channel
  - create_properties
Also fix a log message

jacomago committed:
remove duplicate code for removing channel

jacomago committed:
Add recceiver id

Adds a configuration option to pass a recceiver id with channels
to make cleanOnStart and cleanOnStop only change a single recceivers channels

tynanford commented:
This is a good idea and would address https://github.com/ChannelFinder/recsync/issues/74 nicely. I agree with @simon-ess that using something like the hostname would help reduce manual config and user errors. What do you think @jacomago ?

Could you update cf.conf and demo.conf with the new config option `RECCEIVER_ID` so it's more visible to new users?

jeonghanlee commented on server/recceiver/cfstore.py:
```diff
@@ -502,20 +507,22 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocid, owner, io
         raise defer.CancelledError()
 
-def create_properties(owner, iocTime, hostName, iocName, iocid):
+def create_properties(owner, iocTime, recceiver_id, hostName, iocName, iocid):
```
I think, we should use the same convention here between `receiver_id` and `iocid`. Please select it either convention to keep the similar style, even if we have multiple convention. I don't want to see more conventions in a code.

simon-ess replied:
I think that's good advice; there are already two conventions in that line (`hostName` vs. `iocid`), adding a third is probably overkill :smile:

jacomago replied:
Updated

jacomago committed:
Set hostname as default recceiver id

jacomago committed:
swap to recceiverid instead of using _

jacomago committed:
update cf.conf and demo.conf

jacomago commented:
> This is a good idea and would address #74 nicely. I agree with @simon-ess that using something like the hostname would help reduce manual config and user errors. What do you think @jacomago ?
> 
> Could you update cf.conf and demo.conf with the new config option `RECCEIVER_ID` so it's more visible to new users?

Set the default value to be the hostname and updated the conf files.

simon-ess commented:
LGTM. It would be nice in theory to fix the failing pipelines, but those are pretty clearly a separate task that should be handled otherwise (in particular the Centos7 ones; I haven't seen the other failures before)
